monteCarloTreeSearch.c

```diff
@@ -0,0 +1,334 @@
+		for(i = 0; i < 200; i++)
+		{
+			if(allActions[i] == -1)
+			{
+				break;
+			}
+
+			numOfActions++;
+		}
+		if(numOfActions != 0)
+		{
+
+			for(i = 0; i < numOfActions; i+2)
+			{
+			
+				action[0] = allActions[i];
+				action[1] = allActions[i+ 1];
+				//You need to implement a choose move here and then send that move as action.
+				expand(selected, action);
+				numNewNodes++;
+
+			}
+		}
+		// else
+		// {
+		// 	move[0] = -1;
+		// 	move[1] = -1;
+		// 	flag == -100;
+		//	break;
+		// }
+
+		//3. Simulate the game board on the new node until the game is complete 
+		for(j = 0; j < selected->numChildren; j++)
+		{
+			boardState = selected->board;
+			while(gameContinues > 0)
+			{
+				numOfActions = 0;
+				//get board structure from the node call getmoves and for loop with while loop inside
+				//placePiece()
+				//get the move to be made and play the game until the player wins or loses
+				getMoves(boardState, allActions);
+
+				for(i = 0; i < 200; i++)
+				{
+					if(allActions[i] == -1)
+					{
+						break;
+					}
+
+					numOfActions++;
+				}
+				//make sure that there are still actions to be taken before the game ends
+				if(numOfActions != 0)
+				{
+					//Randomly select a move to make
+					randomValue = numOfActions / 2;
+					randomValue = rand() % randomValue + 1;  //may not be plus 1
+					boardMove[0] = allActions[randomValue];
+					boardMove[1] = allActions[randomValue + 1]; 
+				}
+				else
+				{
+					boardMove[0] = -1;
+					boardMove[1] = -1;
+				}
+				//player cannot make a move
+				if(boardMove[0] == -1 && boardMove[1] == -1)
+				{
+
+					//Check if other player can move
+					flipMove(boardState->whoseMove);
+					pickMove(boardState, nodeLimit, timeLimit, boardMove);
+
+					//determine the winner because other player cannot make a move either
+					if(boardMove[0] == -1 && boardMove[1] == -1)
+					{
+						gameContinues = -10;
+						winner = determineWinner(boardState);
+						if(winner == 0)
+						{
+							selected->numSimulations += 1;
+
+						}
+						else if(winner == 1)
+						{
+							if(boardState->whoseMove == 1)
+							{
+								selected->numWins += 1;
+							}
+							selected->numSimulations += 1;
+
+						}	
+						else if(winner == 2)
+						{
+							if(boardState->whoseMove == 2)
+							{
+								selected->numWins += 1;
+							}
+							selected->numSimulations += 1;
+
+						}
+					}	
+					//Continue the game
+					else
+					{
+						placePiece(boardState, boardMove);
+					}
+				}
+				else
+				{	
+					placePiece(boardState, boardMove);
+				}
+
+			}
+		}
+		//4. Back-propagate results up to the root node once you get the results of your simulation.
+		//This just involves recording progress in the numWins and numSimulations members of nodes and 
+		//following parent up the tree till you hit NULL.
+		
+		 while(newNode->parent !=NULL)
+		 {
+		 	sum = 0;
+		 	numWins = 0;
+			numSims = 0;	
+		 	parentNode = newNode->parent;
```

Comment:
You'll need to sum up the wins/simulations on the newly expanded child nodes of the selected node first, so hold off on this for now.